### ISSUE TITLE
Logic Flaw in Theme Build Process

### ISSUE THREAD

**Issue by robertmclaws** [UXDivers/Grial-UI-Kit-Support#164]
Several threads (#161, etc) have reported this exception:

```
iOS theme colors definition file (ThemeColors.cs) not found in 
project 'D:\Dev\Scratch\Grial\Starter\PikeConnect.Core\PikeConnect.Core.csproj'. 
(project 'D:\Dev\Scratch\Grial\Starter\PikeConnect.Core\PikeConnect.Core.csproj').
```
The exception is being thrown near line 53 of `GrialApplyThemeiOSCommand.cs` in `UXDivers.Daas.Commands.dll`. The code in question is as follows:

```cs
                IFileProjectItem fileByPath = project.GetFileByPath(this.ColorsFileName, true);
                if (fileByPath == null)
                {
                    throw new ExecutionException(this.CommandName, string.Format(SR.iOSThemeColorsFileNotFound, this.ColorsFileName, project.FullPath));
                }
```

So given the resulting exception, the code is looking for `ThemeColors.cs` at `project.FullPath`, which is presently being populated with the PCL project path, not the iOS project path. 

Looking into the `SyncThemeCommand` build task that is calling that project in the first place, it feels like something is missing there. Either `GrialInferPlatformSpecificProjectsCommand()` is not working correctly, (which does not appear to be the case), or the `CurrentProjectPath` is being bound to `GrialApplyThemeiOSCommand` instead of `iOSProjectPath`, which seems more likely, given the expression binding soup in that function.

Given the number of magic strings required for that build task to work, I would seriously consider making those strings `readonly constant`, and then make sure you're passing in the right constant for the binding. 

OR just consider making `SyncThemeTask` A LOT less complicated. I'm not sure what all the binding nonsense is for, when the Task seems like it could be executed a lot more simply by just calling the functions directly and awaiting the results.

**Comment by LeoHere** [comment]
@robertmclaws thanks for the detailed explanation.

Can you try updating to 2.0.43-RC? 

These packages includes several changes to the ```PropagateThemeTask``` like:
- reimplemented theme sync to support Shared projects too
- improved error reporting
- improved logging/feedback

The new version is available in the PreRelease channel. Please, remember to update them across the all your projects.

**Comment by robertmclaws** [comment]
I updated to 2.0.50-RC and that seems to have fixed that issue. Another issue came up after the build actually worked, but I will log that separately.

Thanks!

**Comment by ifegade** [comment]
what is 2.0.43-RC? and how do i go about updating it?

**Comment by LeoHere** [comment]
@ifegade ```2.0.50-RC``` is the version of the ```UXDivers.Artina.*``` Nuget packages that Grial's Starter/Full projects depends on. In order to get the fix for the ```ThemesColors.cs``` you need to update those manually from ```2.0.35``` to ```2.0.50-RC```.

**Comment by jpbrocca** [comment]
Hi all,
this issue is fixed on 2.0.52.0 released today.

For more info, please check:
http://uxdivers.github.io/Grial-UI-Kit-Support/release-notes.html

I am closing this issue to keep things tidy, but feel free to reopen as necessary.
Thanks!

JP
